torchgeo/datasets/spacenet.py

```diff
@@ -191,49 +199,287 @@ def __getitem__(self, index: int) -> Dict[str, Tensor]:
+            archive_path = os.path.join(self.root, collection + ".tar.gz")
+            if os.path.exists(archive_path):
+                print(f"Found {collection} archive")
+                if (
+                    self.checksum
+                    and check_integrity(
+                        archive_path, self.collection_md5_dict[collection]
+                    )
+                    or not self.checksum
+                ):
+                    print("Extracting...")
+                    extract_archive(archive_path)
+                else:
+                    print(f"Collection {collection} is corrupted")
+                    to_be_downloaded.append(collection)
+            else:
+                print(f"{collection} not found")
+                to_be_downloaded.append(collection)
+
+        return to_be_downloaded
+
+    def _download(self, collections: List[str], api_key: Optional[str] = None) -> None:
         """Download the dataset and extract it.
 
         Args:
+            collections: Collections to be downloaded
             api_key: a RadiantEarth MLHub API key to use for downloading the dataset
 
         Raises:
             RuntimeError: if download doesn't work correctly or checksums don't match
         """
-        if self._check_integrity():
-            print("Files already downloaded")
-            return
-
-        download_radiant_mlhub_dataset(self.dataset_id, self.root, api_key)
-        archive_path = os.path.join(self.root, self.foldername + ".tar.gz")
-        if (
-            self.checksum
-            and check_integrity(archive_path, self.md5)
-            or not self.checksum
-        ):
+        for collection in collections:
+            download_radiant_mlhub_collection(collection, self.root, api_key)
+            archive_path = os.path.join(self.root, collection + ".tar.gz")
+            if (
+                not self.checksum
+                or not check_integrity(
+                    archive_path, self.collection_md5_dict[collection]
+                )
+            ) and self.checksum:
+                raise RuntimeError(f"Collection {collection} corrupted")
+
+            print("Extracting...")
             extract_archive(archive_path)
-        else:
-            raise RuntimeError("Dataset corrupted")
+
+
+class SpaceNet1(SpaceNet):
+    """SpaceNet 1: Building Detection v1 Dataset.
+
+    `SpaceNet 1 <https://spacenet.ai/spacenet-buildings-dataset-v1/>`_
+    is a dataset of building footprints over the city of Rio de Janeiro.
+
+    * *Dataset features*:
+
+        * No. of images: 6940 (8 Band) + 6940 (RGB)
+        * No. of polygons: 382,534 building labels
+        * Area Coverage: 2544 sq km
+        * GSD: 1 m (8 band),  50 cm (rgb)
+        * Chip size: 102 x 110 (8 band), 407 x 439 (rgb)
+
+    .. note::
+       Chip size of both imagery can have 1 pixel difference
+
+    * *Dataset format*:
+
+        * Imagery - Worldview-2 GeoTIFFs
+                * 8Band.tif (Multispectral)
```

Comment:
These are indented too far, 4 spaces is probably sufficient

Reply:
Modified